be/src/util/path_util.cpp

```diff
@@ -0,0 +1,88 @@
+// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+#include "util/path_util.h"
+
+// Use the POSIX version of dirname(3). See `man 3 dirname`
+#include <libgen.h>
+
+#include <cstring>
+
+#include <common/logging.h>
+#include "gutil/strings/split.h"
+#include "gutil/strings/stringpiece.h"
+#include "gutil/strings/strip.h"
+#include "env/env.h"
+#include "common/status.h"
+
+using std::string;
+using std::vector;
+using strings::SkipEmpty;
+using strings::Split;
+
+namespace doris {
+namespace path_util {
+
+const string kTmpInfix = ".doristmp";
+
+string join_path_segments(const string& a, const string& b) {
+    CHECK(!a.empty()) << "empty first component: " << a;
+    CHECK(!b.empty() && b[0] != '/')
+            << "second path component must be non-empty and relative: "
+            << b;
```

Comment:
I don't think it's a good idea to CHECK common function's input arguments

Reply:
I will change it to `DCHECK`.

BTW, for file systems that simulate a hierarchical structure, repeated '/' can be problematic.  For example: in some object-storage-services, it may consider `"/a/b"` and `"/a///b"` as different objects. And during the file operation, no error will be reported, so check here to avoid such errors.